tests/jenkins/TestPrintArtifactDownloadUrlsForStaging.groovy

```diff
@@ -6,23 +6,73 @@
  * compatible open source license.
  */
 
-package jenkins.tests
-
+import jenkins.tests.BuildPipelineTest
 import org.junit.*
 
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.core.IsNull.notNullValue
+
+
+class TestPrintArtifactDownloadUrlsForStaging extends BuildPipelineTest implements LibFunctionTester {
+
+    private List artifactFileNames
+    private String uploadPath
 
-class TestPrintArtifactDownloadUrlsForStaging extends BuildPipelineTest {
 
     @Before
     void setUp() {
+
+        this.artifactFileNames = ['dummy_file.tar.gz', 'dummy_file.tar.gz.sig', 'a_dummy_file.tar.gz']
+        this.uploadPath = 'dummy/upload/path'
+
+        this.registerLibTester(new TestPrintArtifactDownloadUrlsForStaging(
+                artifactFileNames: artifactFileNames,
+                uploadPath: uploadPath
+        ))
+
         super.setUp()
-        binding.setVariable('filenamesForUrls', ['dummy_file.tar.gz', 'dummy_file.tar.gz.sig'])
-        binding.setVariable('UPLOAD_PATH', 'dummy/upload/path')
     }
 
     @Test
     void testPrintArtifactDownloadUrlsForStaging() {
+
+        binding.setVariable('filenamesForUrls', artifactFileNames)
+        binding.setVariable('UPLOAD_PATH', uploadPath)
+
         super.testPipeline("tests/jenkins/jobs/PrintArtifactDownloadUrlsForStaging_Jenkinsfile")
+
+    }
+
+    void configure(helper, bindings){
+    }
```

Comment:
We aren't mocking anything? it makes the function under test seem less useful

Reply:
The configure is empty because we don't need any variables set for this function.